### PR TITLE
Python extension

### PR DESCRIPTION
**Pull request by adamheins**
This is a proposal for changing the Python bindings such that they're written as a C++ extension, one benefit of which is that it can be installed in the usual way with `python setup.py install` (and thus can be readily included in other projects without copying code). I've matched the API from the original implementation, and the tests are the same except for the type of exception raised for passing the wrong argument.

Let me know if you think this would be a useful change. I'm happy to make any changes to this code, and if we decide to go forward with this PR I can clean up the commit history before merging.

### CODE REVIEW

File: python/setup.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    "miniball",
+    sources=["miniball_python.cpp"],
+)
+
+setup(
+    name="miniball",
+    version="1.0",
+    description="Library to find the smallest enclosing ball of points",
+    author="Martin Kutz",
+    author_email="kutz@math.fu-berlin.de",
```
Comment (hbf) [inline python/setup.py:17]: Can you change the author to Kaspar Fischer, Bernd Gärtner, and Martin Kutz; as for the email, Martin Kutz is sadly not with us anymore so I suggest to remove it (the URL points to this GitHub project, which should be a good way to get help).

**Review by hbf — APPROVED** [review]
@adamheins Thanks for the contribution! Great catch also spotting and removing unnecessary Python dependencies.

**Comment by hbf** [comment]
> I'm happy to make any changes to this code, and if we decide to go forward with this PR I can clean up the commit history before merging.

FYI I switched the project settings to only offer “Squash and merge”.

File: python/setup.py (hunk 1 above)
Comment (adamheins) [reply python/setup.py]: Done!

**Comment by hbf** [comment]
Thanks again. Happy to add you also to the [README](https://github.com/hbf/miniball#authors--acknowledgements) if you’d like that, feel free to make a PR.

**Pull request merged**